### ISSUE TITLE
Broken AWS Authentication in version 3.9

### ISSUE THREAD

**Issue by dahateb** [hashicorp/terraform-provider-vault#1624]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.
1.0.11 & 1.2.4

### Affected Resource(s)
Please list the resources as a list, for example:
- provider "vault"

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```
provider "vault" {
  address = "https://vault"

  auth_login {
    path   = "auth/aws/login"
    method = "aws"
    parameters = {
      role       = "our_role"
      sts_region = "us-east-1"
    }
  }
}
```

### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.
```
2022-10-06T08:30:30.957Z [INFO]  provider.terraform-provider-vault_v3.9.0_x5: 2022/10/06 08:30:30 [DEBUG] Vault API Request Details:
---[ REQUEST ]---------------------------------------
PUT /v1/auth/aws/login HTTP/1.1
Host: vault.eu-central-1.platform.tier-services.io
User-Agent: Go-http-client/1.1
Content-Length: 174
X-Vault-Request: true
X-Vault-Token: hmac-sha256:383d71dac9e61890f6312e7496ae09dae61cdc4d9a3d37ab6a99940fee3462a4
Accept-Encoding: gzip


2022-10-06T08:30:31.036Z [INFO]  provider.terraform-provider-vault_v3.9.0_x5: 2022/10/06 08:30:31 [DEBUG] Vault API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 400 Bad Request
Content-Length: 60
Cache-Control: no-store
Content-Type: application/json
Date: Thu, 06 Oct 2022 08:30:31 GMT
Strict-Transport-Security: max-age=15724800; includeSubDomains

```

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Expected Behavior
What should have happened?

### Actual Behavior
What actually happened?
```
╷
│ Error: Error making API request.
│ 
│ URL: PUT https://vault.eu-central-1.platform.tier-services.io/v1/auth/aws/login
│ Code: 400. Errors:
│ 
│ * didn't supply required authentication values
│ 
│   with provider["registry.terraform.io/hashicorp/vault"],
│   on providers.tf line 11, in provider "vault":
│   11: provider "vault" {
│ 
╵
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`

**Comment by benashz** [comment]
We'll be issuing a patch release today that should address this issue.

**Comment by benashz** [comment]
Thanks again for reporting this issue.

**Comment by vinay-gopalan** [comment]
A fix for this was released in the latest patch version [v3.9.1](https://github.com/hashicorp/terraform-provider-vault/releases/tag/v3.9.1). Please let us know if you face any other issues, and thanks once again for bringing this to our attention!
